src/hooks/useActiveLocale.ts

```diff
@@ -14,6 +16,22 @@ function parseLocale(maybeSupportedLocale: string): SupportedLocale | undefined
   )
 }
 
+/**
+ * Returns the supported locale read from the url query parameter (lng=)
+ */
+function urlLocale(): SupportedLocale | undefined {
+  const search = location.search || location.hash.match(/\?.*/)?.[0]
+  const parsed = search && search.length > 1 ? parse(search, { parseArrays: false, ignoreQueryPrefix: true }) : {}
+  return typeof parsed.lng === 'string' ? parseLocale(parsed.lng) : undefined
+}
+
+/**
+ * Returns the supported locale read from the user state (redux)
+ */
+function userLocale(): SupportedLocale | undefined {
+  return initialState.user?.userLocale ?? undefined
```

Comment:
won't that always be `null`?

Reply:
It may be set, if you've explicitly set a locale in the past.

For example, if you navigate to `example.com?lng=fr-FR`, then `fr-FR` will be set in your `localStorage` (through redux), and this will be in your `initialState` (which is loaded from localStorage as part of initializing redux).